### PR TITLE
[stablehlo] verify stablehlo backend contract

### PR DESCRIPTION
**Pull request by penguin-wwy**

### CODE REVIEW

**Comment by penguin-wwy** [comment]
Please review this pr? @stellaraccident @qingyunqu

**Review by stellaraccident — APPROVED** [review]
I don't know much about the stablehlo backend but seems ok to me based on what I do know.

File: lib/Dialect/TorchConversion/Transforms/VerifyStablehloBackendContract.cpp [hunk 1]
```diff
@@ -47,13 +46,21 @@ class VerifyStablehloBackendContractPass
...
-    target.addDynamicallyLegalOp<shape::ShapeOfOp>(opHasLegalTypes);
 
     target.addLegalDialect<chlo::ChloDialect>();
     target.addLegalDialect<stablehlo::StablehloDialect>();
     target.addLegalDialect<tensor::TensorDialect>();
     target.addLegalDialect<arith::ArithDialect>();
+    target.addLegalDialect<math::MathDialect>();
+    target.addLegalDialect<shape::ShapeDialect>();
+    auto moduleOp = getOperation();
+    RewritePatternSet patterns(context);
```
Comment (qingyunqu) [inline lib/Dialect/TorchConversion/Transforms/VerifyStablehloBackendContract.cpp:57]: seems `patterns` doesn't have any pattern. Is there any another method to check the dialect legality?

Comment (penguin-wwy) [reply lib/Dialect/TorchConversion/Transforms/VerifyStablehloBackendContract.cpp:57]: It is implemented in the same way in other verification passes (e.g. VerifyLinalgOnTensorsBackendContract)

Additionally, I have reviewed the git history of this file ( https://github.com/llvm/torch-mlir/commit/711646d0951a50af7ea30012cbf588c3103fc0ab#diff-4675df3b49f8dcf2eb2bbe27e96d125e5ef37db8802cd240258f6659ff2f0113L57 ), this method is how it was originally done, but this part was deleted for testing reasons when mhlo dialect migrate conversion to stablehlo.

Comment (qingyunqu) [reply lib/Dialect/TorchConversion/Transforms/VerifyStablehloBackendContract.cpp:57]: Thanks! LGTM

**Review by qingyunqu — APPROVED** [review]

**Pull request merged**